src/main/java/uk/nhs/tis/sync/job/PersonCurrentPmSyncJobTemplate.java

```diff
@@ -0,0 +1,158 @@
+package uk.nhs.tis.sync.job;
+
+import com.google.common.base.Stopwatch;
+import com.transformuk.hee.tis.tcs.service.model.Person;
+import java.math.BigInteger;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Query;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import uk.nhs.tis.sync.event.JobExecutionEvent;
+
+/**
+ * abstract template for Jobs which sync data when current programmeMembership changes.
+ */
+public abstract class PersonCurrentPmSyncJobTemplate implements RunnableJob {
+
+  protected static final int DEFAULT_PAGE_SIZE = 5000;
+  protected static final int FIFTEEN_MIN = 15 * 60 * 1000;
+  protected static final String FULL_SYNC_DATE_STR = "ANY";
+  protected static final String NO_DATE_OVERRIDE = "NONE";
```

Comment:
As part of design discussions, I thought you didn't want to include any of this in the (Reval) Current Programme update job.

It's not clear when you decided to lay some of the groundwork for supplying dates to either job but I'm happy with what you've done here :smile: